flytekit-jackson/src/main/java/org/flyte/flytekit/jackson/LiteralMapSerializer.java

```
@@ -0,0 +1,263 @@
+        return;
+
+      case MAP:
+        gen.writeFieldName("literal");
+        gen.writeObject(Literal.Kind.MAP);
+        gen.writeFieldName("type");
+        gen.writeObject(literalTypeMap.get(key).mapValueType().simpleType());
+        gen.writeFieldName("value");
+        gen.writeStartObject();
+
+        for (Map.Entry<String, Literal> entry : value.map().entrySet()) {
+          gen.writeFieldName(entry.getKey());
+          gen.writeStartObject();
+          serializeLiteral(entry.getKey(), entry.getValue(), gen, serializers, literalTypeMap);
+          gen.writeEndObject();
+        }
+
+        gen.writeEndObject();
+        return;
+
+      case COLLECTION:
+        gen.writeFieldName("literal");
+        gen.writeObject(Literal.Kind.COLLECTION);
+        gen.writeFieldName("type");
+        gen.writeObject(literalTypeMap.get(key).collectionType().simpleType());
+        gen.writeFieldName("value");
+        gen.writeStartArray();
+
+        for (Literal element : value.collection()) {
+          gen.writeStartObject();
+          serializeLiteral(key, element, gen, serializers, literalTypeMap);
+          gen.writeEndObject();
+        }
+
+        gen.writeEndArray();
+        return;
+    }
+
+    throw new AssertionError("Unexpected Literal.Kind: [" + value.kind() + "]");
+  }
+
+  public void serializeScalar(
+      Scalar value,
+      JsonGenerator gen,
+      SerializerProvider serializers,
+      Map<String, LiteralType> literalTypeMap)
+      throws IOException {
+    switch (value.kind()) {
+      case PRIMITIVE:
+        gen.writeFieldName("scalar");
+        gen.writeObject(Scalar.Kind.PRIMITIVE);
+        serializePrimitive(value.primitive(), gen);
+        return;
+
+      case GENERIC:
+        gen.writeFieldName("scalar");
+        gen.writeObject(Scalar.Kind.GENERIC);
+        serializeGeneric(value.generic(), gen, serializers, literalTypeMap);
+        return;
+
+      case BLOB:
```

Should we throw unsupported here?